app/src/hooks/useTooltipOffset.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+export function useTooltipOffset(cardRef) {
+  const [offset, setOffset] = useState<[number, number]>([0, -130]);
+  const getOffset = () => {
+    if (cardRef.current) {
+      const image = cardRef.current.children[0] as HTMLElement;
```

Comment:
I'm glad you were able to extract the hook and got it to work as we discussed. I did not consider having to pass a ref to every card component but then only reading the first one in the children array. If this is the case, I think you can get away with passing the ref only to the first `DCCard` component in the map function and not pass it to the rest. I don't think it's a big issue rendering- or computation-wise but minor optimization.

Reply:
I agree and also I think that change should happen in this ticket, not here https://newyorkpubliclibrary.atlassian.net/browse/DR-3191